chromium_src/ui/views/controls/button/md_text_button.cc

```diff
@@ -71,143 +71,37 @@ constexpr float kLoadingOpacity = 0.75f;
-      {.background_color = std::nullopt,
-       .border_color = std::nullopt,
-       .text_color = gfx::kColorTextDisabled}},
-     {{ui::ButtonStyle::kTonal, ColorScheme::kDark,
-       ButtonState::STATE_DISABLED},
-      {.background_color = std::nullopt,
-       .border_color = std::nullopt,
-       .text_color = gfx::kColorTextDisabledDark}},
-
-     {{ui::ButtonStyle::kText, ColorScheme::kLight, ButtonState::STATE_NORMAL},
-      {.background_color = std::nullopt,
-       .border_color = std::nullopt,
-       .text_color = gfx::kColorTextSecondary}},
-     {{ui::ButtonStyle::kText, ColorScheme::kDark, ButtonState::STATE_NORMAL},
-      {.background_color = std::nullopt,
-       .border_color = std::nullopt,
-       .text_color = gfx::kColorTextSecondaryDark}},
-     {{ui::ButtonStyle::kText, ColorScheme::kLight, ButtonState::STATE_HOVERED},
-      {.background_color = SkColorSetA(gfx::kColorGray60, 0xFF * 0.05),
-       .border_color = std::nullopt,
-       .text_color = views::internal::kColorTextSecondaryBlack}},
-     {{ui::ButtonStyle::kText, ColorScheme::kDark, ButtonState::STATE_HOVERED},
-      {.background_color = SkColorSetA(gfx::kColorGray20, 0xFF * 0.1),
-       .border_color = std::nullopt,
-       .text_color = views::internal::kColorTextSecondaryDarkWhite}},
-     {{ui::ButtonStyle::kText, ColorScheme::kLight,
-       ButtonState::STATE_DISABLED},
-      {.background_color = std::nullopt,
-       .border_color = std::nullopt,
-       .text_color = gfx::kColorTextDisabled}},
-     {{ui::ButtonStyle::kText, ColorScheme::kDark, ButtonState::STATE_DISABLED},
-      {.background_color = std::nullopt,
-       .border_color = std::nullopt,
-       .text_color = gfx::kColorTextDisabledDark}}});
-
-class BraveTextButtonHighlightPathGenerator
-    : public views::HighlightPathGenerator {
- public:
-  BraveTextButtonHighlightPathGenerator() = default;
-  BraveTextButtonHighlightPathGenerator(
-      const BraveTextButtonHighlightPathGenerator&) = delete;
-  BraveTextButtonHighlightPathGenerator& operator=(
-      const BraveTextButtonHighlightPathGenerator&) = delete;
-
-  // HighlightPathGenerator
-  SkPath GetHighlightPath(const views::View* view) override;
-};
+static constexpr auto kButtonThemes =
+    base::MakeFixedFlatMap<MdTextButtonStyleKey, ButtonStyle>({
+        // Kind=Filled
+        {{ui::ButtonStyle::kProminent, ButtonState::STATE_NORMAL},
+         {.background_color = nala::kColorButtonBackground}},
+        {{ui::ButtonStyle::kProminent, ButtonState::STATE_HOVERED},
+         {.background_color = nala::kColorPrimary60,
+          .dark_background_color = nala::kColorPrimary50}},
+
+        // Kind=Outline
+        {{ui::ButtonStyle::kDefault, ButtonState::STATE_NORMAL},
+         {.border_color = nala::kColorDividerInteractive,
+          .text_color = nala::kColorTextInteractive}},
+        {{ui::ButtonStyle::kDefault, ButtonState::STATE_HOVERED},
+         {.background_color = nala::kColorNeutral20,
```

Comment:
It seems hover bg color is different with figma.

With this PR:
<img width="308" height="130" alt="image" src="https://github.com/user-attachments/assets/fba15bf3-69cc-4378-a13b-75743a99dcc3" />

but in figma (https://www.figma.com/design/g8Z0q6TMPYDq6zXh9Y7LWD/%F0%9F%8C%8E-Universal?node-id=278-12158&p=f&m=dev):

also focused border color seems slightly different?
<img width="382" height="230" alt="image" src="https://github.com/user-attachments/assets/84853f43-4098-4c43-b3a4-bb52dbd905ee" />

Reply:
Ah interesting - so I actually copied these colors directly from the Nala button - looks like our fallback for when we don't support color-mix isn't quite right for the outline button background.

This won't match perfectly but I've updated it to use primitive-primary-10 which is a lot closer (the button in Nala uses color-mix)

Reply:
Interestingly for me the focused color looks right

Reply:
Ah, ok. focus ring seems matched with storybook.